ui/src/data-products/element-list/continuous-geometry-search.tsx

```diff
@@ -80,30 +73,27 @@ const ContinuousGeometrySearch = ({
     function getVisibleErrorsByProp(prop: keyof ContinuousSearchParameters) {
         return state.committedFields.includes(prop)
             ? state.validationErrors
-                  .filter((error) => error.field == prop)
-                  .map((error) => t(`data-products.element-list.search.${error.reason}`))
+                .filter((error) => error.field == prop)
+                .map((error) => t(`data-products.element-list.search.${error.reason}`))
             : [];
     }
 
     function hasErrors(prop: keyof ContinuousSearchParameters) {
         return getVisibleErrorsByProp(prop).length > 0;
     }
 
-    useLoader(() => {
-        return (
-            !state.searchParameters.locationTrack || hasErrors('searchGeometries')
-                ? Promise.resolve([])
-                : getLocationTrackElements(
-                      state.searchParameters.locationTrack.id,
-                      selectedElementTypes(state.searchParameters.searchGeometries),
-                      validTrackMeterOrUndefined(state.searchParameters.startTrackMeter),
-                      validTrackMeterOrUndefined(state.searchParameters.endTrackMeter),
-                  )
-        ).then((elements) => {
-            setElements(elements ?? []);
```

Comment:
Itse bugi oli tässä: callbackia käytettiin suoraan promisen then-blokissa, jolloin race-tilanteessa sitä kutsuttiin kaikille pyynnöille siinä järjestyksessä kun ne valmistuu. Näin aiemmin aloitettu mutta pidempään kestänyt pyyntö saattoi jäädä viimeisenä voimaan -> epäsynkka